packages/botonic-react/src/components/multichannel/multichannel-utils.js

```diff
@@ -78,3 +78,35 @@ export const isFacebook = context =>
   context.session &&
   context.session.user &&
   context.session.user.provider == Providers.Messaging.FACEBOOK
+
+export function whatsappMarkdown(text) {
+  const markdownRegex = /(\*\*|__)(.*?)\1|(\*|_)(.*?)\3/g
+
+  const matches = text.match(markdownRegex)
+  if (matches) {
+    const matchesResult = matches.map(match => {
+      if (match.startsWith('**')) {
+        return match.replaceAll('**', '*')
+      }
+      if (match.startsWith('__')) {
+        return match.replaceAll('__', '*')
+      }
+      if (match.startsWith('*')) {
+        return match.replaceAll('*', '_')
+      }
+      // if( match.startsWith('_')) {
+      //   return match.replaceAll('_', '_')
+      // }
+      return match
+    })
+    let textWhatsapp = ''
+    for (let i = 0; i < matches.length; i++) {
+      if (i === 0) {
+        textWhatsapp = text.replaceAll(matches[i], matchesResult[i])
+      }
+      textWhatsapp = textWhatsapp.replaceAll(matches[i], matchesResult[i])
```

Comment:
we should see how we can do it in an easier way. I would ask GPT, but i’m not sure if it’s the best option, let’s talk with somebody of the team to help you simplifying it

Reply:
I looked for this solution with GPT and at first it gave me a solution that didn't work, in the end I modified the code to this one that does work. Surely it can be done better